### ISSUE TITLE
Error importing angular in ng1-to-ng2

### ISSUE THREAD

**Issue by lukashambsch** [ui-router/angular-hybrid#20]
There seems to be an issue importing angular in the ng1-to-ng2.js file. I'm using systemjs with the following config file:

```
(function (global) {
 System.config({
    baseUrl: 'app',
    paths: {
      // paths serve as alias
      'npm:': 'node_modules/'
    },
    meta: {
      'npm:angular/angular': {format: 'global', exports: 'angular'},
      'ui-router-ng2': {format: 'cjs'},
    },
    // map tells the System loader where to look for things
    map: {
      // our app is within the app folder
      app: 'app/ng2Components',
      // angular bundles
      '@angular/core': 'npm:@angular/core/bundles/core.umd.js',
      '@angular/common': 'npm:@angular/common/bundles/common.umd.js',
      '@angular/compiler': 'npm:@angular/compiler/bundles/compiler.umd.js',
      '@angular/platform-browser': 'npm:@angular/platform-browser/bundles/platform-browser.umd.js',
      '@angular/platform-browser-dynamic': 'npm:@angular/platform-browser-dynamic/bundles/platform-browser-dynamic.umd.js',
      '@angular/http': 'npm:@angular/http/bundles/http.umd.js',
      '@angular/router': 'npm:@angular/router/bundles/router.umd.js',
      '@angular/forms': 'npm:@angular/forms/bundles/forms.umd.js',
      '@angular/upgrade': 'npm:@angular/upgrade/bundles/upgrade.umd.js',
      // other libraries
      'rxjs':                      'npm:rxjs',
      'angular':                   'npm:angular/angular.js',
      'angular-ui-router':         'npm:angular-ui-router/release/angular-ui-router.js',
      'angular-in-memory-web-api': 'npm:angular-in-memory-web-api',
      'ui-router-ng2':             'npm:ui-router-ng2/_bundles/ui-router-ng2.js',
      'ui-router-ng1-to-ng2':      'npm:ui-router-ng1-to-ng2/ng1-to-ng2.js',
    },
    // packages tells the System loader how to load when no filename and/or no extension
    packages: {
      app: {
        main: './main.js',
        defaultExtension: 'js'
      },
      rxjs: {
        defaultExtension: 'js'
      },
      'angular-in-memory-web-api': {
        main: './index.js',
        defaultExtension: 'js'
      }
    }
  });
})(this);
```

Here is the full error I am getting:

```
(index):47 Error: (SystemJS) angular.module is not a function
    TypeError: angular.module is not a function
        at Object.module.exports (http://localhost:3000/node_modules/ui-router-ng1-to-ng2/ng1-to-ng2.js:86:34)
        at __webpack_require__ (http://localhost:3000/node_modules/ui-router-ng1-to-ng2/ng1-to-ng2.js:36:30)
        at c (http://localhost:3000/node_modules/ui-router-ng1-to-ng2/ng1-to-ng2.js:56:18)
        at eval (http://localhost:3000/node_modules/ui-router-ng1-to-ng2/ng1-to-ng2.js:59:10)
        at Object.eval (http://localhost:3000/app/ng2Components/ng2app.module.js:15:30)
    Evaluating http://localhost:3000/app/ng2Components/ng2app.module.js
    Evaluating http://localhost:3000/app/ng2Components/main.js
    Error loading http://localhost:3000/app/ng2Components/main.js
        at Object.module.exports (http://localhost:3000/node_modules/ui-router-ng1-to-ng2/ng1-to-ng2.js:86:34)
        at __webpack_require__ (http://localhost:3000/node_modules/ui-router-ng1-to-ng2/ng1-to-ng2.js:36:30)
        at c (http://localhost:3000/node_modules/ui-router-ng1-to-ng2/ng1-to-ng2.js:56:18)
        at eval (http://localhost:3000/node_modules/ui-router-ng1-to-ng2/ng1-to-ng2.js:59:10)
        at Object.eval (http://localhost:3000/app/ng2Components/ng2app.module.js:15:30)
    Evaluating http://localhost:3000/app/ng2Components/ng2app.module.js
    Evaluating http://localhost:3000/app/ng2Components/main.js
```

The app uses script tags to add all of the angular 1 files, could this be causing the issue?

**Comment by samjetski** [comment]
We had the same issue. It was fixed by deleting our `node_modules` directory and re-running `npm install`. Not sure why though, shouldn't need to do this. For reference, we're using Webpack instead of SystemJS.

**Comment by christopherthielen** [comment]
We've since released version 3.0.0.  For SystemJS users you will have to set up mappings for:

- `@uirouter/core`
- `@uirouter/angular`
- `@uirouter/angularjs`
- `@uirouter/angular-hybrid`
- `@uirouter/rx`

UI-Router expects the `angular` object to be available to the module loader.  You may need to do some shimming or otherwise tell SystemJS that `angular` should come from `window.angular`.  https://github.com/systemjs/systemjs/blob/master/docs/config-api.md#user-content-meta

This isn't really my area of expertise.  I have had troubles with SystemJS for production apps, and thus have migrated to webpack.  The [updated sample app](https://github.com/ui-router/sample-app-ng1-to-ng2) also uses webpack.

If you're still having troubles, please re-open with further details and I'll try to help if I can.
